### PR TITLE
[JENKINS-43651] allow recognizing Maven through non-system-wide defined MAVEN_HOME env var

### PR DESCRIPTION
**Pull request by repolevedavaj**
allow recognizing Maven through non-system-wide defined MAVEN_HOME env var (fixes JENKINS-43651)

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: jenkins-plugin/src/test/resources/org/jenkinsci/plugins/pipeline/maven/docker/NonMavenJavaContainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM jenkins/java:d93654cc6239
+
+RUN apt-get remove maven -y maven
```
Comment (bguerin) [inline jenkins-plugin/src/test/resources/org/jenkinsci/plugins/pipeline/maven/docker/NonMavenJavaContainer/Dockerfile:3]: duplicate `maven`  no ?

**Review by bguerin — CHANGES_REQUESTED** [review]
Great catch and great PR, with unit test, many thanks

Would you just mind adding a unit test to ensure maven is also found from environment, when installed directly on the agent, without using tools at all ?

File: jenkins-plugin/src/test/resources/org/jenkinsci/plugins/pipeline/maven/docker/NonMavenJavaContainer/Dockerfile (hunk 1 above)
Comment (repolevedavaj) [reply jenkins-plugin/src/test/resources/org/jenkinsci/plugins/pipeline/maven/docker/NonMavenJavaContainer/Dockerfile]: yes, you are right - i removed the second one

**Comment by repolevedavaj** [comment]
@bguerin I added additional test-cases to `WithMavenStepTest` and took the chance to refactor the class a bit. I hope that's OK. Let met know whether there's something else to improve :)

**Comment by bguerin** [comment]
@repolevedavaj awesome, thanks a lot !

**Review by bguerin — APPROVED** [review]

**Comment by aheritier** [comment]
great contribution @repolevedavaj 
Thanks

**Pull request merged**